src/robot/parsing/model/statements.py

```diff
@@ -946,12 +946,15 @@ def variable(self):
 
     def validate(self):
         as_token = self.get_token(Token.AS)
-        if as_token:
-            if as_token is not self.tokens[-2]:
-                self.errors += ("EXCEPT's AS marker must be second to last.",)
-            var = self.tokens[-1].value
-            if not is_scalar_assign(var):
-                self.errors += (f"EXCEPT's AS variable '{var}' is invalid.",)
+        if as_token:                
+            var = self.get_token(Token.VARIABLE)
+            
+            if var is None:
+                self.errors += ("EXCEPT's AS expects a variable.",)
+            elif next((v for v in self.tokens[self.tokens.index(var) + 1:] if v.type not in Token.NON_DATA_TOKENS), None):
+                self.errors += (f"EXCEPT's AS can only have one variable.",)
+            elif not is_scalar_assign(var.value):
+                self.errors += (f"EXCEPT's AS variable '{var.value}' is invalid.",)
```

Comment:
This looks a bit complicated. If lexing wouldn't be changed at all, validation could look something like this:

```python
as_token = self.get_token(Token.AS)
if as_token:
    variables = self.get_token(Token.VARIABLE)
    if not variables:
        self.errors += ("EXCEPT's AS requires variable.",)
    elif len(variables) > 1:
        self.errors += ("EXCEPT's AS marker accepts only one variable.",)
    elif not is_scalar_assign(variables[0].value):
        self.errors + (f"EXCEPT's AS variable '{variables[0].value}' is invalid.",)
```

Compared to the original validation, there are now separate errors for `AS` having not variables and for it having more than one. Thus, as I commented the updated test, it would be good to have separate acceptance tests for these situations as well.